### PR TITLE
fix(querier): use most specific timerange available

### PR DESCRIPTION
**Pull request by JohnnyQQQQ**
This PR ensures that we select the right timerange for a query, based on the following criteria:

- If a query has its own timerange, use that.
- If a query don't have it's own timerange
  - Try first to get the timerange from the parent request
  - Otherwise use the default `0`, `0` 

This is important as backend services like alerting don't set any timerange in the request itself but only on the queries. This caused all queries to run on a 6 hour range, which caused high latency and resource drain.

### CODE REVIEW

**Review by ryantxu — APPROVED** [review]
LGTM -- good catch!

File: pkg/registry/apis/query/parser.go [hunk 1]
```diff
@@ -189,6 +181,19 @@ func (p *queryParser) parseRequest(ctx context.Context, input *query.QueryDataRe
...
 }
 
+func getTimeRangeForQuery(parentTimerange, queryTimerange *data.TimeRange) data.TimeRange {
+	if queryTimerange != nil && queryTimerange.From != "" && queryTimerange.To != "" {
+		return *queryTimerange
+	}
+	if parentTimerange != nil && parentTimerange.To != "" && parentTimerange.From != "" {
+		return *parentTimerange
+	}
+	return data.TimeRange{
```
Comment (toddtreece) [inline pkg/registry/apis/query/parser.go:191]: is it valid to make a request without a time range currently? i would expect an error

Comment (JohnnyQQQQ) [reply pkg/registry/apis/query/parser.go:191]: Without any is also a bit strange to me, I guess that's something we could change as part of a new API version

Comment (ryantxu) [reply pkg/registry/apis/query/parser.go:191]: missing values are currently sent on as zero:
https://github.com/grafana/grafana-plugin-sdk-go/blob/main/backend/gtime/time_range.go#L64

Comment (toddtreece) [reply pkg/registry/apis/query/parser.go:191]: is `now-6h` coming from existing code? to me it makes more sense to error or to leave it empty if the query and parent request don't have a time range set. setting it to `now-6h` might be a good default for some data sources, or result in an expensive query for others

Comment (ryantxu) [reply pkg/registry/apis/query/parser.go:191]: @JohnnyQQQQ -- can you update the PR so missing time gets a range 0-0 like it did before?

Comment (JohnnyQQQQ) [reply pkg/registry/apis/query/parser.go:191]: What would 0-0 be in terms or timerange string? `now`, `now`?

Comment (JohnnyQQQQ) [reply pkg/registry/apis/query/parser.go:191]: I guess I can set them to `0` and `0`.

Comment (toddtreece) [reply pkg/registry/apis/query/parser.go:191]: i think leaving them empty would also do it

Comment (JohnnyQQQQ) [reply pkg/registry/apis/query/parser.go:191]: Just tested that locally so it should be fine

```
		val := gtime.NewTimeRange("0", "0")
		tr := &backend.TimeRange{
			From: val.GetFromAsTimeUTC(),
			To:   val.GetToAsTimeUTC(),
		}
		require.Equal(t, tr.Duration(), time.Duration(0))
```

**Review by toddtreece — APPROVED** [review]

**Pull request merged**